File: src/main/java/corp/soyummy/recipes/Recipe.java
package corp.soyummy.recipes;

import lombok.Data;
import lombok.Builder;

import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@Document(collection = "recipes")
public class Recipe {
    @Id
    private String id;

    @Field("title")
    private String title;

    @Field("category")
    private String category;

    @Field("area")
    private String area;

    @Field("instructions")
    private String instructions;

    @Field("description")
    private String description;

    @Field("thumb")
    private String thumbnail;

    @Field("preview")
    private String preview;

    @Field("time")
    private String time;

    @Field("popularity")
    private Integer popularity;

    @Field("favorites")
    private List<String> favorites;

    @Field("likes")
    private List<String> likes;

    @Field("youtube")
    private String youtube;

    @Field("tags")
    private List<String> tags;

    @Field("createdAt")
    private LocalDateTime createdAt;

    @Field("updatedAt")
    private LocalDateTime updatedAt;

    @Field("ingredients")
    private List<RecipeIngredient> ingredients;
}

File: src/main/kotlin/com/soyummy/recipes/Recipe.kt
package com.soyummy.recipes

import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.index.TextIndexed
import java.time.LocalDateTime

@Document(collection = "recipes")
data class Recipe(
    @Id
    val id: String? = null,

    @Field("title")
    @Indexed
    @TextIndexed(weight = 3f)
    val title: String,

    @Field("category")
    @Indexed
    val category: String,

    @Field("area")
    @Indexed
    val area: String = "Unknown",

    @Field("instructions")
    val instructions: String,

    @Field("description")
    @TextIndexed(weight = 1f)
    val description: String,

    @Field("thumb")
    val thumbnail: String,

    @Field("preview")
    val preview: String,

    @Field("time")
    val time: String,

    @Field("popularity")
    @Indexed(direction = org.springframework.data.mongodb.core.index.IndexDirection.DESCENDING)
    val popularity: Int = 0,

    @Field("favorites")
    val favorites: List<String> = emptyList(),

    @Field("likes")
    val likes: List<String> = emptyList(),

    @Field("youtube")
    val youtube: String,

    @Field("tags")
    @Indexed
    val tags: List<String>? = emptyList(),

    @Field("ingredients")
    val ingredients: List<RecipeIngredient>? = emptyList(),

    @Field("createdAt")
    val createdAt: LocalDateTime,

    @Field("updatedAt")
    val updatedAt: LocalDateTime
)